otx/core/data/adapter/segmentation_dataset_adapter.py

```diff
@@ -52,3 +66,104 @@ def get_otx_dataset(self) -> DatasetEntity:
+
+        Self-SL for semantic segmentation using DetCon uses pseudo masks as labels,
+        but Datumaro cannot load this custom data structure because it is not in Datumaro format.
+        So, it is required to manually load and set annotations.
+
+        Args:
+            train_data_roots (Optional[str]): Path for training data.
+            pseudo_mask_dir (str): Directory to save pseudo masks. Defaults to "detcon_mask".
+
+        Returns:
+            DatumaroDataset: Datumaro Dataset
+        """
+        assert train_data_roots
+        logger = get_logger()
+        logger.warning(f"Please check if {train_data_roots} is data roots only for images, not annotations.")
+
+        dataset = {}
+        dataset[Subset.TRAINING] = DatumaroDataset.import_from(train_data_roots, format="image_dir")
+        self.is_train_phase = True
+
+        # Load pseudo masks
+        img_dir = None
+        total_labels = []
+        flag_create_mask = False
+        for item in dataset[Subset.TRAINING]:
+            img_path = item.media.path
+            if img_dir is None:
+                # Get image directory
+                img_dir = train_data_roots.split("/")[-1]
+            pseudo_mask_path = img_path.replace(img_dir, pseudo_mask_dir)
+            if pseudo_mask_path.endswith(".jpg"):
+                pseudo_mask_path = pseudo_mask_path.replace(".jpg", ".png")
+
+            if not os.path.isfile(pseudo_mask_path):
+                # Create pseudo mask
+                flag_create_mask = True
+                pseudo_mask = self.create_pseudo_masks(item.media.data, pseudo_mask_path)  # type: ignore
+            else:
+                # Load created pseudo mask
+                pseudo_mask = cv2.imread(pseudo_mask_path, cv2.IMREAD_GRAYSCALE)
+
+            # Set annotations into each item
+            annotations = []
+            labels = np.unique(pseudo_mask)
+            for label_id in labels:
+                if label_id not in total_labels:
+                    # Stack label_id to save dataset_meta.json
+                    total_labels.append(label_id)
+                annotations.append(
+                    Mask(image=CommonSemanticSegmentationBase._lazy_extract_mask(pseudo_mask, label_id), label=label_id)
+                )
+            item.annotations = annotations
+
+        pseudo_mask_roots = train_data_roots.replace(img_dir, pseudo_mask_dir)  # type: ignore
+        if flag_create_mask:
+            # Save dataset_meta.json for newly created pseudo masks
+            meta = {"label_map": {i + 1: f"target{i+1}" for i in range(max(total_labels) + 1)}}
+            with open(os.path.join(pseudo_mask_roots, "dataset_meta.json"), "w", encoding="UTF-8") as f:
+                json.dump(meta, f, indent=4)
+
+        # Make categories for pseudo masks
+        label_map = parse_meta_file(os.path.join(pseudo_mask_roots, "dataset_meta.json"))
+        dataset[Subset.TRAINING]._data._categories = make_categories(label_map)
```

Comment:
I think it is wrong to use the protected attribute in class.

Reply:
Done.